lib/intercepted_request_router.js

```diff
@@ -305,6 +305,8 @@ class InterceptedRequestRouter {
     )
 
     if (matchedInterceptor) {
+      matchedInterceptor.requests.push(req)
```

Comment:
`ClientRequest` does not have a friendly API (think about getting the body). Maybe we should transform it to a `Request` instance?

Reply:
how?